### PR TITLE
Fix getMaxId won't get the right max id before GC

### PR DESCRIPTION
**Pull request by jiaqizho**
### What problem does this PR solve?

Issue Number: ref #3594 

Problem Summary:
- When we del the page id
- In getMaxId won't now current page_id have been deleted

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/PageDirectory.cpp [hunk 1]
```diff
@@ -889,10 +889,31 @@ PageId PageDirectory::getMaxId(NamespaceId ns_id) const
...
+                break;
+            }
+
+            // Find the last valid one
+            if (iter->second->getEntry(UINT64_MAX - 1) != std::nullopt)
+            {
+                return iter->first.low;
+            }
+
+            // Current entries is deleted. There are no entries before current.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectory.cpp:907]:
```suggestion
            // Current entry is deleted and there are no entries before it.
```

File: dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp [hunk 2]
```diff
@@ -2057,6 +2057,36 @@ try
...
+        dir->apply(std::move(edit));
+    }
+    ASSERT_EQ(dir->getMaxId(TEST_NAMESPACE_ID), 1);
+
+    {
+        PageEntriesEdit edit;
+        edit.del(1);
+        dir->apply(std::move(edit));
+    }
+    ASSERT_EQ(dir->getMaxId(TEST_NAMESPACE_ID), 0);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp:2086]: call `dir->gcInMemEntries` after line 2086 and check the result of getMaxId again

File: dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp [hunk 3]
```diff
@@ -2057,6 +2057,36 @@ try
 }
 CATCH
 
+TEST_F(PageDirectoryTest, GetMaxId2)
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp:2060]:
```suggestion
TEST_F(PageDirectoryTest, GetMaxIdAfterDelete)
```

File: dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp [hunk 4]
```diff
@@ -2057,6 +2057,36 @@ try
 }
```
Comment (lidezhu) [inline dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp:2057]:
```suggestion
    {
        PageEntriesEdit edit;
        edit.del(buildV3Id(medium, 320));
        dir->apply(std::move(edit));
        ASSERT_EQ(dir->getMaxId(medium), 300);
    }

    {
        PageEntriesEdit edit;
        edit.del(buildV3Id(medium, 300));
        dir->apply(std::move(edit));
        ASSERT_EQ(dir->getMaxId(medium), 0);
    }
}
```

Comment (lidezhu) [reply dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp:2057]: This is to test delete with more than one `ns_id` in PageDirectory.

Comment (jiaqizho) [reply dbms/src/Storages/Page/V3/tests/gtest_page_directory.cpp:2071]: done

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Comment by jiaqizho** [comment]
need revert

**Pull request merged**